### PR TITLE
Fixes some Amazon price parsing cases and adds a bunch of new tests

### PR DESCRIPTION
**Pull request by vegaro**
There are some issues:
- Floating point issues
- 1.000,00 is not correctly parsed

### CODE REVIEW

File: feature/amazon/src/main/java/com/revenuecat/purchases/amazon/storeProductConversions.kt [hunk 1]
```diff
@@ -71,18 +72,16 @@ internal fun String.parsePriceUsingRegex(): Float? {
...
+            // Most currencies but Dirhan use 2 decimals. Amazon doesn't support Dirhan at the moment
+            price = if (split.last().length == 3) {
+                price.replace(".", "").replace(",", "")
             } else {
-                // 1,00
-                price.replace(",", ".")
+                val intPart = split.dropLast(1).joinToString("")
+                "$intPart.${split.last()}"
             }
         }
```
Comment (vegaro) [inline feature/amazon/src/main/java/com/revenuecat/purchases/amazon/storeProductConversions.kt:86]: I like this implementation much more. I also think it's easier to understand

File: feature/amazon/src/main/java/com/revenuecat/purchases/amazon/storeProductConversions.kt [hunk 2]
```diff
@@ -67,27 +68,25 @@ internal data class Price(
...
-            val numberOfCommas = price.length - price.replace(",", "").length
-            price = if (price.contains(".") || numberOfCommas > 1) {
-                // 1,000,000.00
-                price.replace(",", "")
+        var price = dirtyPrice.replace(" ", "")
+        val split = price.split(".", ",")
+        if (split.size != 1) {
+            // Assuming all prices we get have 2 decimal points
+            // Most currencies but Dirhan use 2 decimals. Amazon doesn't support Dirhan at the moment
+            price = if (split.last().length == 3) {
```
Comment (aboedo) [inline feature/amazon/src/main/java/com/revenuecat/purchases/amazon/storeProductConversions.kt:80]: do we have any way to identify right-to-left orientation?

Comment (vegaro) [reply feature/amazon/src/main/java/com/revenuecat/purchases/amazon/storeProductConversions.kt:80]: We can check the locale of the device, but prices are not formatted according to the locale of the device. For example, I tested a Live App Testing build with a user in Amazon ES (the device was in US English), and the price was coming as "1,10 €".

In any case, I don't think right-to-left matters for numbers right? I believe prices would still be represented the same. Currency might go on the right instead but the number itself is not right-to-left.

Comment (aboedo) [reply feature/amazon/src/main/java/com/revenuecat/purchases/amazon/storeProductConversions.kt:80]: You're right! TIL: they're [represented the same](https://qr.ae/pGLRjH). That's good to know! 


> In Arabic, it's the natural ordering from right to left: least significant digit first, most significant last.
> 
> Historically, the Western numbering system is derived from the Hindu-Arabic numbering system; popularized by Muslim mathematicians like [Muḥammad ibn Mūsā al-Khwārizmī](https://en.wikipedia.org/wiki/Mu%E1%B8%A5ammad_ibn_M%C5%ABs%C4%81_al-Khw%C4%81rizm%C4%AB). When these books were translated from Arabic, however, Europeans who had a left-to-right language, still maintained the Arabic numbering order. Why Europeans did this nobody knowes.

File: feature/amazon/src/main/java/com/revenuecat/purchases/amazon/storeProductConversions.kt [hunk 3]
```diff
@@ -67,27 +68,25 @@ internal data class Price(
...
+                price.replace(".", "").replace(",", "")
```
Comment (beylmk) [inline feature/amazon/src/main/java/com/revenuecat/purchases/amazon/storeProductConversions.kt:81]:
```suggestion
// the price has no decimal parts, so we can simply take the number
                price.replace(".", "").replace(",", "")
```

File: feature/amazon/src/main/java/com/revenuecat/purchases/amazon/storeProductConversions.kt (hunk 1 above)
Comment (beylmk) [reply feature/amazon/src/main/java/com/revenuecat/purchases/amazon/storeProductConversions.kt:86]: agreed

**Review by beylmk — APPROVED** [review]
🤑

**Pull request merged**